### PR TITLE
Add tutorial on output validation

### LINKED ISSUE #355: No output validation mentioned in validation tutorial
http://hapijs.com/tutorials/validation
Happy to contribute content if I get a :+1:

### PR DESCRIPTION
**Pull request by dylancwood**
Closes #355

Please let me know if you'd like anything changed.

### CODE REVIEW

File: lib/tutorials/validation.md [hunk 1]
```diff
@@ -117,3 +117,49 @@ You may validate incoming headers as well, with a `validate.headers` parameter.
 ### Payload parameters
 
 Also valid is the `validate.payload` parameter, which will validate payload data sent to a route by the user. It works exactly the same way as query parameters, in that if you validate one key, you must validate them all.
+
+## Output
+
+Hapi can also validate responses before they are sent back to the client.
```
Comment (mtharrison) [inline lib/tutorials/validation.md]: Hapi => hapi

File: lib/tutorials/validation.md [hunk 2]
```diff
@@ -117,3 +117,49 @@ You may validate incoming headers as well, with a `validate.headers` parameter.
...
+
+If a response does not pass the response validation, the client will receive a `500: internal server error` response instead.
+
+Output validation is useful for ensuring that your API is serving data that is consistent with its documentation/contract.
+Additionally, plugins like [hapi-swagger](https://github.com/glennjones/hapi-swagger) and [lout](https://github.com/hapijs/lout) can use the response-validation schemas to automatically document each
+endpoint's output format, thus ensuring that your documentation is always up to date.
+
+Here is an example route configuration that returns a list of books:
+
+```json
```
Comment (mtharrison) [inline lib/tutorials/validation.md]: should be javascript not json

Comment (dylancwood) [inline lib/tutorials/validation.md]: doh! Thanks

File: lib/tutorials/validation.md [hunk 3]
```diff
@@ -117,3 +117,49 @@ You may validate incoming headers as well, with a `validate.headers` parameter.
...
+    config: {
+        response: {
+            schema: Joi.array().items(bookSchema);
+        }
+    }
+});
+
+```
+
+If any books do not match the `bookSchema` exactly, then Hapi will respond with a `500` error code. The error response will *not* indicate the reason for the error.
```
Comment (mtharrison) [inline lib/tutorials/validation.md]: Hapi => hapi again

**Comment by mtharrison** [comment]
Maybe worth mentioning `sample` option too:

> sample - the percent of response payloads validated (0 - 100). Set to 0 to disable all validation. Defaults to 100 (all response payloads).

File: lib/tutorials/validation.md [hunk 4]
```diff
@@ -117,3 +117,49 @@ You may validate incoming headers as well, with a `validate.headers` parameter.
...
+});
+
+server.route({
+    method: 'GET',
+    path: '/books',
+    handler: function (request, reply) {
+        getBooks(function (err, books) {
+          if (err) {
+            return reply(Boom.wrap(err));
+          } else {
```
Comment (mtharrison) [inline lib/tutorials/validation.md]: Nitpicky maybe, but no need for the `else` here as you're returning early above

**Comment by devinivy** [comment]
Cool that you wrote this up!  I think before publishing this information, it's worth noting additional info about different fail actions for output validation– sometimes a 500 isn't appropriate– and the ability to sample a certain percentage of responses.

**Comment by dylancwood** [comment]
Thank you for the quick review @devinivy. I've addressed each of your comments in my latest commit. Please let me know if there is anything else.

**Comment by nlf** [comment]
this looks great! thanks for the quick reviews @mtharrison and @devinivy and thank *you* @dylancwood for writing this up. with their feedback in it this looks ready to go to me

**Comment by devinivy** [comment]
:+1: :+1: Looks great on the site.

File: lib/tutorials/validation.md [hunk 5]
```diff
@@ -117,3 +117,92 @@ You may validate incoming headers as well, with a `validate.headers` parameter.
...
+    datePublished: Joi.date().iso()
+});
+
+server.route({
+    method: 'GET',
+    path: '/books',
+    config: {
+        handler: function (request, reply) {
+            getBooks(function (err, books) {
+              if (err) {
```
Comment (gergoerdosi) [inline lib/tutorials/validation.md]: Spaces should be consistent for indentation.

Comment (dylancwood) [inline lib/tutorials/validation.md]: Good catch. I was lazy and didn't change my tabspace to 4. I'll fix it and submit a new PR.

**Pull request merged**